### PR TITLE
docs: add content on introspection warnings

### PR DESCRIPTION
**Pull request by ruheni**
## Changes
Documents introspection stopgaps (warnings for unsupported features) — what they are and how to get around them

### CODE REVIEW

**Review by janpio — COMMENTED** [review]
This is in no way a replacement for #4798, which actually describes how the stopgap then can be used to replace the missing feature. It probably also does not close https://github.com/prisma/team-orm/issues/61 because of that.

Besides that, good step to document possible introspection warnings.

File: content/200-concepts/100-components/04-introspection.mdx [hunk 1]
```diff
@@ -357,4 +357,51 @@ Introspecting only a subset of your database schema is [not yet officially suppo
...
+
+  - [Check Constraints](https://github.com/prisma/prisma/issues/3388) (MySQL + PostgreSQL)
+  - [Exclusion Constraints](https://github.com/prisma/prisma/issues/17514)
+  - [MongoDB $jsonSchema](https://github.com/prisma/prisma/issues/8135)
+
+- From version 4.16.0:
+
+  - [Expression indexes](https://github.com/prisma/prisma/issues/2504)
+
+We will expand this to more database features. You can find the list of features we intend to support on [GitHub (labeled with `topic:database-functionality`)](https://github.com/prisma/prisma/issues?q=is%3Aopen+label%3A%22topic%3A+database-functionality%22+label%3Ateam%2Fschema+sort%3Aupdated-desc+).
```
Comment (janpio) [inline content/200-concepts/100-components/04-introspection.mdx]: We do not "intend" to support these with warnings or proper implementations, those are just issues from the same area.

File: content/200-concepts/100-components/04-introspection.mdx [hunk 2]
```diff
@@ -357,4 +357,51 @@ Introspecting only a subset of your database schema is [not yet officially suppo
...
+
+- From version 4.14.0:
+
+  - [Check Constraints](https://github.com/prisma/prisma/issues/3388) (MySQL + PostgreSQL)
+  - [Exclusion Constraints](https://github.com/prisma/prisma/issues/17514)
+  - [MongoDB $jsonSchema](https://github.com/prisma/prisma/issues/8135)
+
+- From version 4.16.0:
+
+  - [Expression indexes](https://github.com/prisma/prisma/issues/2504)
```
Comment (janpio) [inline content/200-concepts/100-components/04-introspection.mdx]: Add link to release notes

**Comment by ruheni** [comment]
> This is in no way a replacement for https://github.com/prisma/docs/pull/4798, which actually describes how the stopgap then can be used to replace the missing feature. It probably also does not close https://github.com/prisma/team-orm/issues/61 because of that.

I think we can have one example to highlight the workaround — covering all of them would not make sense. I would be happy to expand the workaround section we have.

File: content/200-concepts/100-components/04-introspection.mdx [hunk 3]
```diff
@@ -357,4 +357,51 @@ Introspecting only a subset of your database schema is [not yet officially suppo
...
+
+- From version [4.14.0](https://github.com/prisma/prisma/releases/tag/4.14.0):
+
+  - [Check Constraints](https://github.com/prisma/prisma/issues/3388) (MySQL + PostgreSQL)
+  - [Exclusion Constraints](https://github.com/prisma/prisma/issues/17514)
+  - [MongoDB $jsonSchema](https://github.com/prisma/prisma/issues/8135)
+
+- From version [4.16.0](https://github.com/prisma/prisma/releases/tag/4.16.0):
+
+  - [Expression indexes](https://github.com/prisma/prisma/issues/2504)
```
Comment (nikolasburk) [inline content/200-concepts/100-components/04-introspection.mdx]: why all the extra line breaks? 😅 

```suggestion
- From version [4.13.0](https://github.com/prisma/prisma/releases/tag/4.13.0):
  - [Partitioned tables](https://github.com/prisma/prisma/issues/1708)
  - [PostgreSQL Row Level Security](https://github.com/prisma/prisma/issues/12735)
  - [Index sort order, `NULLS FIRST` / `NULLS LAST`](https://github.com/prisma/prisma/issues/15466)
  - [CockroachDB row-level TTL](https://github.com/prisma/prisma/issues/13982)
  - [Comments](https://github.com/prisma/prisma/issues/8703)
  - [PostgreSQL deferred constraints](https://github.com/prisma/prisma/issues/8807)
- From version [4.14.0](https://github.com/prisma/prisma/releases/tag/4.14.0):
  - [Check Constraints](https://github.com/prisma/prisma/issues/3388) (MySQL + PostgreSQL)
  - [Exclusion Constraints](https://github.com/prisma/prisma/issues/17514)
  - [MongoDB $jsonSchema](https://github.com/prisma/prisma/issues/8135)
- From version [4.16.0](https://github.com/prisma/prisma/releases/tag/4.16.0):
  - [Expression indexes](https://github.com/prisma/prisma/issues/2504)
```

File: content/200-concepts/100-components/04-introspection.mdx [hunk 4]
```diff
@@ -357,4 +357,51 @@ Introspecting only a subset of your database schema is [not yet officially suppo
...
+
+1. Create a draft migration:
+
+   ```terminal
+   npx prisma migrate dev --create-only
+   ```
+
+2. Add the SQL that adds the feature surfaced in the warnings.
+
+3. Apply the draft migration to your database:
```
Comment (nikolasburk) [inline content/200-concepts/100-components/04-introspection.mdx:403]: same here

```suggestion
1. Create a draft migration:
   ```terminal
   npx prisma migrate dev --create-only
   ```
2. Add the SQL that adds the feature surfaced in the warnings.
3. Apply the draft migration to your database:
```

File: content/200-concepts/100-components/04-introspection.mdx [hunk 5]
```diff
@@ -357,4 +357,46 @@ Introspecting only a subset of your database schema is [not yet officially suppo
...
+
+   ```terminal
+   npx prisma migrate dev
+   ```
```
Comment (nikolasburk) [inline content/200-concepts/100-components/04-introspection.mdx]:
```suggestion
1. Create a draft migration:
   ```terminal
   npx prisma migrate dev --create-only
   ```
2. Add the SQL that adds the feature surfaced in the warnings.
3. Apply the draft migration to your database:
   ```terminal
   npx prisma migrate dev
   ```
```

File: content/200-concepts/100-components/04-introspection.mdx [hunk 6]
```diff
@@ -357,4 +357,36 @@ Introspecting only a subset of your database schema is [not yet officially suppo
...
+
+You can find the list of features we intend to support on [GitHub (labeled with `topic:database-functionality`)](https://github.com/prisma/prisma/issues?q=is%3Aopen+label%3A%22topic%3A+database-functionality%22+label%3Ateam%2Fschema+sort%3Aupdated-desc+).
+
+### Workaround for introspection warnings for unsupported features
+
+If you are using a relational database and either one of the above features listed in the previous section:
+
+1. Create a draft migration:
+   ```terminal
+   npx prisma migrate dev --create-only
```
Comment (nikolasburk) [inline content/200-concepts/100-components/04-introspection.mdx:392]:
````suggestion
1. Create a draft migration:
   ```terminal
   npx prisma migrate dev --create-only
   ```
2. Add the SQL that adds the feature surfaced in the warnings.
3. Apply the draft migration to your database:
   ```terminal
   npx prisma migrate dev
   ```
````

**Comment by nikolasburk** [comment]
The changes look useful but I'm also not sure if they close the related issues, so I updated the comment to remove that. We can still close the issues manually and/or expand on the current changes in another PR 🙏

**Comment by janpio** [comment]
This PR is completely unreviewed and should not have been merged.

**Comment by nikolasburk** [comment]
I reviewed the docs and thought they looked good and certainly felt like a helpful addition to the docs. What are the problems you see with this PR?

**Pull request merged**